benchmarks/src/main/java/io/grpc/benchmarks/driver/LoadClient.java

```diff
@@ -0,0 +1,495 @@
+import java.util.concurrent.locks.LockSupport;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Implements the client-side contract for the load testing scenarios.
+ */
+class LoadClient {
+
+  private static final Logger LOG = Logger.getLogger(LoadClient.class.getName());
+  private ByteBuf genericRequest;
+
+  Control.ClientConfig config;
+  AbstractIntegerDistribution distribution;
+  final AtomicBoolean shutdown = new AtomicBoolean();
+  final int threadCount;
+
+  ManagedChannel[] channels;
+  BenchmarkServiceGrpc.BenchmarkServiceBlockingStub[] blockingStubs;
+  BenchmarkServiceGrpc.BenchmarkServiceStub[] asyncStubs;
+  Recorder recorder;
+  private ExecutorService fixedThreadPool;
+  private Messages.SimpleRequest simpleRequest;
+
+  LoadClient(Control.ClientConfig config) throws Exception {
+    LOG.log(Level.INFO, "Client Config \n" + config.toString());
+    this.config = config;
+    // Create the channels
+    channels = new ManagedChannelImpl[config.getClientChannels()];
+    for (int i = 0; i < config.getClientChannels(); i++) {
+      channels[i] =
+          Utils.newClientChannel(Transport.NETTY_EPOLL,
+              Utils.parseSocketAddress(config.getServerTargets(i % config.getServerTargetsCount())),
+              config.hasSecurityParams(),
+              config.hasSecurityParams() && config.getSecurityParams().getUseTestCa(),
+              config.hasSecurityParams()
+                  ? config.getSecurityParams().getServerHostOverride() :
+                  null,
+              true,
+              Utils.DEFAULT_FLOW_CONTROL_WINDOW,
+              false);
+    }
+
+    // Create a stub per channel
+    if (config.getClientType() == Control.ClientType.ASYNC_CLIENT) {
+      asyncStubs = new BenchmarkServiceGrpc.BenchmarkServiceStub[channels.length];
+      for (int i = 0; i < channels.length; i++) {
+        asyncStubs[i] = BenchmarkServiceGrpc.newStub(channels[i]);
+      }
+    } else {
+      blockingStubs = new BenchmarkServiceGrpc.BenchmarkServiceBlockingStub[channels.length];
+      for (int i = 0; i < channels.length; i++) {
+        blockingStubs[i] = BenchmarkServiceGrpc.newBlockingStub(channels[i]);
+      }
+    }
+
+    // Determine no of threads
+    threadCount = config.getClientType() == Control.ClientType.SYNC_CLIENT
+        ? config.getOutstandingRpcsPerChannel() * config.getClientChannels() :
+        config.getAsyncClientThreads();
+    // Use a fixed sized pool of daemon threads.
+    fixedThreadPool = Executors.newFixedThreadPool(threadCount,
+        new DefaultThreadFactory("client-worker", true));
+
+    // Create the load distribution
+    if (config.getLoadParams().getClosedLoop() != null) {
+      distribution = new EnumeratedIntegerDistribution(new int[]{0}, new double[]{1});
+    } else if (config.getLoadParams().getPoisson() != null) {
+      new PoissonDistribution(config.getLoadParams().getPoisson().getOfferedLoad() / threadCount);
```

Comment:
I think the proto is confusing. The field should be called `poissonProcess` or something, so that it's not confused with a poisson distribution. I assume this should be done in the C repo. U think that's still possible at this point?